pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go

```diff
@@ -159,3 +149,19 @@ func (r *Reconciler) newConfigForInMemoryChannel(imc *v1.InMemoryChannel) (*mult
 		},
 	}, nil
 }
+
+func (r *Reconciler) deleteFunc(obj interface{}) {
+	if acc, err := kmeta.DeletionHandlingAccessor(obj); err == nil {
+		if imc, ok := acc.(*v1.InMemoryChannel); ok {
+			// If the IMC has been deleted just make sure the handler is removed.
+			if !imc.GetDeletionTimestamp().IsZero() {
```

Comment:
I don't think this will be true on this path because the object we get is the last copy the informer saw, and without a finalizer this won't be set.